diff --git a/include/configs/cv180x-asic.h b/include/configs/cv180x-asic.h
index 67d1a0e98..6f83b0ef3 100644
--- a/include/configs/cv180x-asic.h
+++ b/include/configs/cv180x-asic.h
@@ -10,24 +10,10 @@
 
 #include <../../../board/cvitek/cv180x/cv180x_reg.h>
 
-/* defined in cvipart.h */
-#undef CONFIG_ENV_OFFSET
-#undef CONFIG_ENV_OFFSET_REDUND
-#undef CONFIG_ENV_SIZE
-#undef CONFIG_ENV_IS_IN_SPI_FLASH
-#undef CONFIG_ENV_IS_IN_MMC
-#undef CONFIG_ENV_IS_IN_NAND
-#undef CONFIG_ENV_SECT_SIZE
-
 /* cvi_board_memmap.h is generated from build/boards/{CHIP_ARCH}/{BOARD}/memmap.py */
 #include "cvi_board_memmap.h"
-/* partition definitions header which is created by mkcvipart.py */
-/* please do not modify header manually */
-#include "cvipart.h"
-#include "cvi_panels/cvi_panel_diffs.h"
 
-// defined in this .h
-#undef CONFIG_BOOTCOMMAND
+#define CONFIG_ENV_SIZE 0x20000
 
 #if defined(__aarch64__)
 #define CONFIG_ARMV8_SWITCH_TO_EL1
@@ -325,13 +311,13 @@
 				"mmc read ${uImage_addr} ${BOOT_PART_OFFSET} ${BOOT_PART_SIZE} ;"		\
 				UBOOT_VBOOT_BOOTM_COMMAND
 	#endif
-
-#else
-	/* define your environment */
-	#define CONFIG_BOOTCOMMAND ""
-
 #endif /* CONFIG_USE_DEFAULT_ENV */
 
+#ifndef CONFIG_EXTRA_ENV_SETTINGS
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	"fdt_addr_r=" __stringify(CVIMMAP_UIMAG_ADDR) "\0"
+#endif
+
 #define CVI_SPL_BOOTAGRS \
 	PARTS " "  \
 	ROOTARGS " " \
